src/libraries/System.Private.CoreLib/src/System/IO/WindowsFileStreamStrategy.cs

```diff
@@ -40,6 +43,7 @@ internal WindowsFileStreamStrategy(SafeFileHandle handle, FileAccess access)
             // Note: Cleaner to set the following fields in ValidateAndInitFromHandle,
             // but we can't as they're readonly.
             _access = access;
+            _share = FileStream.DefaultShare;
```

Comment:
instead of using `FileStream.DefaultShare` here you can add `FileShare share` to the `ctor` arguments and pass `FileStream.DefaultShare` from `FileStream`. We are already doing similar thing:

https://github.com/dotnet/runtime/blob/72a4e94f84a1017dc8c780397dacd7781a82fd6d/src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs#L106